lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -169,19 +131,29 @@ impl GelfDeserializer {
                 if key == "_id" {
                     continue;
                 }
+                // per GELF spec, Additional field names must be prefixed with an underscore
+                if !key.starts_with('_') {
+                    return Err(format!(
+                        "'{}' field is invalid. \
+                                       Additional field names must be prefixed with an underscore.",
+                        key
+                    )
+                    .into());
+                }
                 // per GELF spec, Additional field names must be characters dashes or dots
                 if !VALID_FIELD_REGEX.is_match(key) {
-                    if key.starts_with('_') {
-                        return Err(format!("'{}' field is invalid. \
-                                           Additional field names must be prefixed with an underscore.", key).into());
-                    } else {
-                        return Err(format!("'{}' field contains invalid characters. Field names may \
-                                           contain only letters, numbers, underscores, dashes and dots.", key).into());
-                    }
+                    return Err(format!("'{}' field contains invalid characters. Field names may \
+                                       contain only letters, numbers, underscores, dashes and dots.", key).into());
                 }
 
-                let vector_val: value::Value = val.into();
-                log.insert(path!(key.as_str()), vector_val);
+                // per GELF spec, Additional field values must be either strings or numbers
+                if val.is_string() || val.is_number() {
+                    let vector_val: value::Value = val.into();
+                    log.insert(path!(key.as_str()), vector_val);
+                } else {
+                    return Err(format!("'The value type for field {}' is an invalid type. Additional field values \
+                                       should be either strings or numbers.", key).into());
```

Comment:
```suggestion
                    return Err(format!("The value type for field {} is an invalid type. Additional field values \
                                       should be either strings or numbers.", key).into());
```

The extra `'`s stood out to me.


I was hoping maybe we could include the type that was invalid here too, but I'm not seeing anything readily accessible on [serde_json](https://docs.serde.rs/serde_json/index.html)::[value](https://docs.serde.rs/serde_json/value/index.html)::[Value](https://docs.serde.rs/serde_json/value/enum.Value.html#).

You _could_ do a `match` here to create custom error messages based on the type.

Reply:
good catch, I made a change to provide error message based on actual type.